### ISSUE TITLE
Nested Websocket errors on error returned from the connection

### ISSUE THREAD

**Issue by special-character** [pusher/pusher-js#464]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
The error returned from the error callback gets a heavily nested error that is a `WebSocketError`

ex:
```
pusher.connection.bind(
  'error',
  (rawError) => {
    ...
  }
)
```

The error returned:
```
{
      type: WebSocketError,
      error: {
        type: WebSocketError,
        error: {
          type: WebSocketError,
          message: "The op coudn't be completedd. Socket is not connected"
        }
      }
    }
```

**If the current behavior is a bug, please provide the steps to reproduce and
if possible a minimal demo of the problem via https://jsfiddle.net or similar.**
This is hard to debug but a search on the `WebSocketError` shows lots of places where errors are involved. I think some might be thrown and wrapped.

https://github.com/pusher/pusher-js/search?q=WebSocketError&unscoped_q=WebSocketError

**What is the expected behavior?**
Feels like this should be a standard error. I couldn't find a type that matched with this error.

**Which versions of Pusher, and which browsers / OS are affected by this issue?
Did this work in previous versions of Pusher? If so, which?**
This is specifically happening for `pusher-js/react-native`

Thanks for the project!

**Comment by leesio** [comment]
@special-character I think I've been able to recreate this:

```
2020-07-07 11:48:43.526 [info][tid:com.facebook.react.JavaScript] 'got error from connection', { type: 'WebSocketError',
  error: 
   { type: 'WebSocketError',
     error: { type: 'WebSocketError', error: 'this is an example error' } } }
```

![Screenshot 2020-07-07 at 11 48 23](https://user-images.githubusercontent.com/7124883/86769984-e8d97d80-c047-11ea-9e5d-c10d09f6b370.png)

Can you confirm this is the behaviour you're seeing?

I think this should just be:

```
{ type: 'WebSocketError', error: 'this is an example error' }
```
Can you confirm that that's what you're expecting?

**Comment by special-character** [comment]
@leesio this is what I am seeing and I did expect it to be just one websocket error

**Comment by leesio** [comment]
I think something like this should be fine: 
(note: draft PR as I haven't updated tests)

https://github.com/pusher/pusher-js/pull/476

I can't think of a scenario where the error emitted by the `TransportConnection` or the `Connection` won't be wrapped sensibly. I could be wrong though, so I'll double check this tomorrow.

**Comment by leesio** [comment]
Hey @special-character I just released [v7.0.0](https://github.com/pusher/pusher-js/releases/tag/v7.0.0) which includes this fix. Please let me know if this solves your problem.
